compiler/src/org.graalvm.compiler.replacements.amd64/src/org/graalvm/compiler/replacements/amd64/AMD64GraphBuilderPlugins.java

```diff
@@ -171,6 +179,30 @@ private static void registerMathPlugins(InvocationPlugins plugins, AMD64 arch, B
             registerRound(r, "ceil", RoundingMode.UP);
             registerRound(r, "floor", RoundingMode.DOWN);
         }
+
+        if (arch.getFeatures().contains(CPUFeature.FMA)) {
+            registerFMA(r);
+        }
+    }
+
+    private static void registerFMA(Registration r) {
+        // TODO: have a way to get config.useFMAIntrinsics flag check here.
```

Comment:
I need to have access to config.useFMAIntrinsics here. Is there a shortcut that I can use?

Reply:
I am confused here. I tried to import org.graalvm.compiler.hotspot.GraalHotSpotVMConfig to be able to use it in AMD64GraphBuilderPlugins, but it does not seem to be possible, probably because of the package this file is in? I am not sure whether using another class is good from a design perspective. FMA intrinsics are technically a Math method substitution. Why is there a need for two different classes? Isn't there a way to access the config from this class?

Reply:
You can pass a `boolean useFMAIntrinsics` to the `AMD64GraphBuilderPlugins` constructor. It is called from `org.graalvm.compiler.hotspot.amd64.AMD64HotSpotBackendFactory.createGraphBuilderPlugins` which is HotSpot specific code and thus has access to `GraalHotSpotVMConfig`.

Reply:
I can do this. However, is it something we want to do? I am concerned that it is not a scalable solution, because we need to pass flags independently. If moving forward, math or other method substitutions require Hotspot flags access, we will need to pass all the flags. What do you think? Would it be, for example, possible to make the Hotspot config accessible directly from any plugin class?

Reply:
No, we cannot make HotSpotVMConfig accessible to non-hotspot classes otherwise SVM cannot use these classes. If we start to accumulate a bunch such flags, we can create a `AMD64GraphBuilderPlugins.Params` helper classes to pass them.